Allow user to define a truncation symbol

JosephTLyons opened the pull request:
This allows the user to define a `truncationSymbol`, so that they can append some text to the end of their truncated string:

`this would be an example of the truncated symbol on text that is being trunca...`, where `truncationSymbol = ...`

In some cases, being able to include this might help the user understand the truncation is intentional.  I'm using this action in a workflow that publishes to Discord and I think it would help the users know that the truncation is done on purpose and not just an accidental thing.

One change that might be controversial is consildating the two separate strings (`stringToTruncate` and `acceptableString`) into one (simply `text). so that its easier to follow around in the newly added logic. I can revert this if you'd rather use a few strings to track the different versions.

JosephTLyons committed:
Allow user to define a truncation symbol

JosephTLyons committed:
Run prettier

JosephTLyons commented:
I just realized I didn't update the documentation and there is a question about if the symbols in the truncation string should be elligible to be removed when the `removeDanglingCharacters` option is enabled or if they should be off limits.  I'll wait to hear back from you if this PR is something you'd accept or not before pushing any futher on either of those. :)

strumwolf commented:
Thanks for the PR.

My reasoning on having two separate strings `stringToTruncate` and `accepatableString` is I prefer the idea that `stringToTruncate` as the incoming string and `acceptableString` as the mutated string. Having said that I am fine having the one variable text.

As for the `truncationSybmol` and the `removeDanglingCharacters` option, I think it needs to be one or the other. If `truncationSymbol` is provided, it supersedes `removeDanglingCharacters`. This will allow the addition of the `truncationSymbol` as the last substring and there would be no need to remove any dangling characters.

JosephTLyons commented:
Hey @strumwolf, do you think there should be a hard error if both settings (`truncationSymbol` and `removeDanglingCharacters `) are used, or that the action should just run and only take the `truncationSymbol` into consideration?

JosephTLyons committed:
Fix typos

JosephTLyons committed:
Don't abide by the `removeDanglingCharacters` setting when supplying `truncationSymbol`

JosephTLyons committed:
Delete the INPUT_TRUNCATIONSYMBOL env var after each test

JosephTLyons committed:
Update README.md to mention `truncationSymbol`

JosephTLyons committed:
Revert string variable rename

JosephTLyons commented:
I reverted the string rename, added a test and logic to avoid entering the logic when of `removeDanglingCharacters` when also supplying `truncationSymbol`, and added some documentation to the `README.md` file.

JosephTLyons commented:
Is there anything else here that I can do to close this PR, @strumwolf?
